Reduces the job to only looking at objects older than the cutoff date

MikeMcMahon opened the pull request:
On larger databases it can take a long time to grab all of the artifacts to cleanup.  Django QuerySets are lazy until they are iterated over, which means there is the possibility of timeouts (which we ran into w/ the jobs in tower).  This fix allowed us to clean out the database and set the cutoff dates to a reasonable timeframe.  

Signed-off-by: Mike McMahon <mike.mcmahon@wavefront.com>

Commit:
Reduces the job to only looking at objects older than the cutoff date

Signed-off-by: Mike McMahon <mike.mcmahon@wavefront.com>

matburt commented:
My one big concern here is that it doesn't present the delta of items delete vs skipped, which was the original reasoning behind looping over all (which I admit is grossly inefficient). I wonder if we can get that back, in a way, and still pick up this performance improvement.

MikeMcMahon commented:
definitely, we had an excessively large database and the jobs would not complete and completely bomb out.  This was the only way to get it to proceed.  The delta is nice to have, but the jobs run in the background and so don't I don't read them too much.  It may be enough to get count of the objects and take a count of the actual deleted items, which we infer the skipped items, total items, and deleted items.

ghjm commented on awx/main/management/commands/cleanup_jobs.py:
```diff
@@ -109,7 +110,8 @@ def cleanup_ad_hoc_commands(self):
         skipped, deleted = 0, 0
-        for pu in ProjectUpdate.objects.all():
+        project_updates = ProjectUpdate.objects.filter(created__lte=self.cutoff)
+        for pu in project_updates
```
Needs a colon

MikeMcMahon replied:
caught and fixing.  thanks.

Commit:
fixing missing colon and missing variable usage

Signed-off-by: Mike McMahon <mike.mcmahon@wavefront.com>

AlanCoding commented:
> My one big concern here is that it doesn't present the delta of items delete vs skipped, which was the original reasoning behind looping over all (which I admit is grossly inefficient).

I would suggest that we could obtain this by `filter(created__gte=self.cutoff).count()`, and then just inject that information into the statement that logs the information. That would offer the exact same data to the user, but would still get the performance benefit here.

MikeMcMahon commented:
i think it's fair to do that, and then remove the logic relating to items that would be skipped.

cchurch commented on awx/main/management/commands/cleanup_jobs.py:
```diff
@@ -87,7 +87,8 @@ def cleanup_jobs(self):
         skipped, deleted = 0, 0
-        for ad_hoc_command in AdHocCommand.objects.all():
+        ad_hoc_commands = AdHocCommand.objects.filter(created__lte=self.cutoff)
+        for ad_hoc_command in ad_hoc_commands:
```
I'd also suggest adding `.iterator()` to all of these querysets to minimize memory usage.

Commit:
switching to iterator and adding the missed Job cleanup

Signed-off-by: Mike McMahon <mike.mcmahon@wavefront.com>

Commit:
jobs take count of gte cutoff, process only lt cutoff

Signed-off-by: Mike McMahon <mike.mcmahon@wavefront.com>

MikeMcMahon commented:
switched an iterator as well as calling for a count of everything gte and adding that to the count of skipped at the very end.  I also switched to lt for cutoff, since the logic was gte == skipped, otherwise process it.

matburt reviewed (approved):
Looks great! merging....
